test_conformance/spirv_new/test_spirv_14.cpp

```diff
@@ -352,3 +352,162 @@ TEST_SPIRV_FUNC(spirv14_usersemantic_memberdecoratestring)
+
+    clKernelWrapper kernel = clCreateKernel(prog, "copymemory_test", &error);
+    SPIRV_CHECK_ERROR(error, "Failed to create spv kernel");
+
+    std::vector<int> results(6);
+    clMemWrapper dst =
+        clCreateBuffer(context, CL_MEM_READ_WRITE,
+                       results.size() * sizeof(results[0]), nullptr, &error);
+    SPIRV_CHECK_ERROR(error, "Failed to create dst buffer");
+
+    error |= clSetKernelArg(kernel, 0, sizeof(dst), &dst);
+    SPIRV_CHECK_ERROR(error, "Failed to set kernel args");
+
+    size_t global = 1;
+    error = clEnqueueNDRangeKernel(queue, kernel, 1, nullptr, &global, nullptr,
+                                   0, nullptr, nullptr);
+    SPIRV_CHECK_ERROR(error, "Failed to enqueue kernel");
+
+    error = clEnqueueReadBuffer(queue, dst, CL_TRUE, 0,
+                                results.size() * sizeof(results[0]),
+                                results.data(), 0, nullptr, nullptr);
+    SPIRV_CHECK_ERROR(error, "Unable to read destination buffer");
+
+    const int expected = 42;
+    for (auto result : results)
+    {
+        if (result != expected)
+        {
+            log_error("Result mismatch!  Got %d, Wanted %d\n", result,
+                      expected);
+            return TEST_FAIL;
+        }
+    }
+
+    return TEST_PASS;
+}
+
+TEST_SPIRV_FUNC(spirv14_select_composite)
+{
+    constexpr size_t global_size = 16;
+
+    if (!is_spirv_version_supported(deviceID, "SPIR-V_1.4"))
+    {
+        log_info("SPIR-V 1.4 not supported; skipping tests.\n");
+        return TEST_SKIPPED_ITSELF;
+    }
+
+    cl_int error = CL_SUCCESS;
+
+    clProgramWrapper prog;
+    error =
+        get_program_with_il(prog, deviceID, context, "spv1.4/select_struct");
+    SPIRV_CHECK_ERROR(error, "Failed to compile spv program");
+
+    clKernelWrapper kernel = clCreateKernel(prog, "select_struct_test", &error);
+    SPIRV_CHECK_ERROR(error, "Failed to create spv kernel");
+
+    struct TestStruct
+    {
+        cl_int i;
+        cl_float f;
+    };
+
+    std::vector<TestStruct> results(global_size);
+    clMemWrapper dst =
+        clCreateBuffer(context, CL_MEM_READ_WRITE,
+                       results.size() * sizeof(results[0]), nullptr, &error);
+    SPIRV_CHECK_ERROR(error, "Failed to create dst buffer");
+
+    error |= clSetKernelArg(kernel, 0, sizeof(dst), &dst);
+    SPIRV_CHECK_ERROR(error, "Failed to set kernel args");
+
+    error = clEnqueueNDRangeKernel(queue, kernel, 1, nullptr, &global_size,
+                                   nullptr, 0, nullptr, nullptr);
+    SPIRV_CHECK_ERROR(error, "Failed to enqueue kernel");
+
+    error = clEnqueueReadBuffer(queue, dst, CL_TRUE, 0,
+                                results.size() * sizeof(results[0]),
+                                results.data(), 0, nullptr, nullptr);
+    SPIRV_CHECK_ERROR(error, "Unable to read destination buffer");
+
+    const TestStruct expected_even{ 1024, 3.1415f };
+    const TestStruct expected_odd{ 2048, 2.7128f };
```

Comment:
I think "even" and "odd" are mixed up here; perhaps just rename them to "struct_a", "struct_b" to align with the names in the SPIR-V assembly.

Reply:
Good idea - I switched the names to "struct_a" and "struct_b" to match the SPIR-V source.